### DISCUSSION TITLE
Instant scroll behavior in <ScrollRestoration />

### DISCUSSION THREAD

**Discussion by tavoyne** [remix-run/remix#8391] [Proposals]
I use `scroll-behavior: smooth` for smooth scrolling on anchor click. The problem is that it smooths the scroll restoration on page navigation too. This occurs because `<ScrollRestoration />` internally doesn't provide any value for `options.behavior` when calling `scrollTo()`, which relies on the value of `scroll-behavior` by default.

Wouldn't it be better if the `"instant"` behavior was enforced in `<ScrollRestoration />`? I don't see why anyone would want to see the entire new page scrolled from bottom to top with every navigation.

Or, at the very least, we need a prop to customize the behavior:
```tsx
<ScrollRestoration behavior="instant" />
```

Ready to do a PR if you're okay with it.

**Comment by dduupp** [comment]
+1

Alternatively, is there a way to detect the navigation event and temporarily disable the smooth scroll style when it occurs?
